resource_detectors/lib/opentelemetry/resource/detectors/google_cloud_platform.rb

```diff
@@ -38,12 +38,29 @@ def detect # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metr
             resource_attributes[OpenTelemetry::SemanticConventions::Resource::CONTAINER_NAME] = ENV['CONTAINER_NAME']
           end
 
+          if gcp_env.knative?
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_PROVIDER] = 'gcp'
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_ACCOUNT_ID] = gcp_env.project_id
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::FAAS_NAME] = gcp_env.knative_service_id
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::FAAS_VERSION] = gcp_env.knative_service_revision
+            zone = gcp_env.instance_zone
+            resource_attributes[OpenTelemetry::SemanticConventions::Resource::CLOUD_REGION] = get_region zone
```

Comment:
`get_region` here can return nil, so that means there's scenario where users of this detector will receive a [warning message](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/internal.rb#L47-L58) without the ability to resolve it other than patching this code path.  Let's make sure we only set this key if there is a non-nil value.

Reply:
Yes if `get_region` returns nil then that field won't get set. This is handled by below piece of code
`resource_attributes.delete_if { |_key, value| value.nil? || value.empty? }`

Reply:
Ah, you're right.  I missed that.